nixos/modules/services/networking/mosquitto.nix

```diff
@@ -196,37 +189,24 @@ in
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
       serviceConfig = {
-        Type = "forking";
-        User = "mosquitto";
-        Group = "mosquitto";
+        DynamicUser = true;
```

Comment:
Right with the new systemd it automatically chown everything on start.

Reply:
Indeed.